tests/pipeline/test_store.py

```diff
@@ -18,6 +58,87 @@ def test_process_sample_none():
+        )
+    out_dir = os.path.abspath(out_dir)
+    context = PipelineContext(unique_pipeline_name='test pipeline')
+    context.data_dir = out_dir
+    store = _TestSaveDetectionSamples(context=context,
+                                      event_log=logging.getLogger())
+    img = Image.new('RGB', (60, 30), color='red')
+
+    detections = [
+                    {
+                     'label': 'person',
+                     'confidence': np.float32(0.98),
+                     'box': {
+                         'xmin': np.float32(0.1),
+                         'ymin': np.float32(1.1),
+                         'xmax': np.float32(2.1),
+                         'ymax': np.float32(3.1)
+                        }
+                    }
+                ]
+
+    processed_samples = list(store.process_sample(image=img,
+                                                  thumbnail=img,
+                                                  inference_result=detections))
+    assert len(processed_samples) == 1
+    print(processed_samples)
+    img_out = processed_samples[0]['image']
+    assert img_out == img
+    inf = processed_samples[0]['inference_result']
+    print(inf)
+
+    category = inf[0]['label']
+    confidence = inf[0]['confidence']
+    (x0, y0) = inf[0]['box']['xmin'], inf[0]['box']['ymin']
+    (x1, y1) = inf[0]['box']['xmax'], inf[0]['box']['ymax']
+
+    assert category == 'person'
+    assert isinstance(confidence, np.float32)
+    assert isinstance(x0, np.float32)
+    assert isinstance(y0, np.float32)
+    assert isinstance(x1, np.float32)
+    assert isinstance(y1, np.float32)
+    assert store._save_sample_called
+    assert store._inf_result == detections
+    assert store._img_path
+    img_dir = os.path.dirname(os.path.abspath(store._img_path / "../../"))
+    assert img_dir == out_dir
+    out_img = Image.open(store._img_path)
+    print(img_dir)
+    print(store._img_path)
+    assert out_img.mode == 'RGB'
+    assert out_img.size[0] == 60
+    assert out_img.size[1] == 30
+    json_dir = os.path.dirname(os.path.abspath(store._json_path / "../../"))
+    assert json_dir == out_dir
+    print(json_dir)
+    print(store._json_path)
+    with open(store._json_path) as f:
+        json_inf = json.load(f)
+        print(json_inf)
+        img_fname = json_inf['image_file_name']
+        rel_dir = json_inf['rel_dir']
+        img_fpath = os.path.join(out_dir, rel_dir, img_fname)
+        assert img_fpath == str(store._img_path)
+        json_inf_res = json_inf['inference_result']
+        assert len(json_inf_res) == 1
+        json_inf_res = json_inf_res[0]
+        assert json_inf_res['label'] == 'person'
+        assert isinstance(json_inf_res['confidence'], float)
+        assert isinstance(json_inf_res['box']['xmin'], float)
+        assert isinstance(json_inf_res['box']['ymin'], float)
+        assert isinstance(json_inf_res['box']['xmax'], float)
+        assert isinstance(json_inf_res['box']['ymax'], float)
+
```

Comment:
@bhavikapanara let's clean up after the test.  Otherwise it may lead to side effects that affect other tests, github tracking, etc.

See [pytest fixtures](https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code). 
Among other boilerplate fixtures, there is a convenient way to use per test tmp directory.

Reply:
@ivelin Did you mean to delete the created directory (i.e tmp/detections/20210223-140610.026448) after finishing the test?

Reply:
Done

Reply:
OK, this works, but please take a note to move towards pytest fixtures going forward.

Reply:
@bhavikapanara  See how much cleaner the code would look with the pytest fixture for temp dirs:

https://docs.pytest.org/en/latest/reference.html#tmpdir

```
tmpdir
Tutorial: Temporary directories and files

tmpdir()[source]
Return a temporary directory path object which is unique to each test function invocation, created as a sub directory of the base temporary directory.

By default, a new base temporary directory is created each test session, and old bases are removed after 3 sessions, to aid in debugging. If --basetemp is used then it is cleared each session. See The default base temporary directory.

The returned object is a py.path.local path object.


```